src/data/players.ts

```diff
@@ -32,10 +35,43 @@ export const players: Player[] = [
 	}
 ];
 
-export const playersById = players.reduce<{ [key: string]: Player }>(
-	(acc, curr) => ({
-		...acc,
-		[curr.id]: curr
-	}),
-	{}
+function createPlayersStore() {
+	const { subscribe, update: storeUpdate } = writable<Player[]>(initialPlayers);
+
+	const add = (name: string, handicap: number) => {
+		storeUpdate((players) => [...players, { id: uuidv4(), name, tripIds: ['mb2022'], handicap }]);
+	};
+
+	const update = (playerId: string, name: string, handicap: number) =>
+		storeUpdate((players) => {
+			const existingPlayerIndex = players.findIndex((p) => p.id === playerId);
+			const existingPlayer = players[existingPlayerIndex];
+			if (existingPlayer) {
+				const newPlayers = [...players];
+				newPlayers.splice(existingPlayerIndex, 1, {
+					...existingPlayer,
+					name,
+					handicap
+				});
+				return newPlayers;
+			}
+			return players;
+		});
+	return {
+		subscribe,
+		add,
+		update
+	};
+}
+
+export const players = createPlayersStore();
+
+export const playersById = derived(players, ($players) =>
+	$players.reduce<{ [key: string]: Player }>(
+		(acc, curr) => ({
+			...acc,
+			[curr.id]: curr
+		}),
+		{}
+	)
```

Comment:
this is a derived store (so it will stay in sync) to convert the players array into an indexed object.

eg.
```js
players = [
  { id: 'john', name: 'John', handicap: 10 },
  { id: 'alex', name: 'Alex', handicap: 12 },
];

playersById = {
  alex: { id: 'alex', name: 'Alex', handicap: 12 },
  john: { id: 'john', name: 'John', handicap: 11 },
};

// for quick access eg
playersById[john]
```